packages/@storefront/flux-capacitor/src/core/requests/utils.ts

```diff
@@ -59,10 +59,16 @@ namespace RequestHelpers {
       query: Selectors.pastPurchaseQuery(state),
       refinements: Selectors.pastPurchaseSelectedRefinements(state),
       skip: Selectors.pastPurchasePageSize(state) * (Selectors.pastPurchasePage(state) - 1),
-      sort: RequestAdapter.extractSort(Selectors.pastPurchaseSortSelected(state)),
     };
 
-    return <Request>{ ...request, ...overrideRequest };
+    const sort = RequestAdapter.extractPastPurchaseSort(
+      <any>overrideRequest.sort || Selectors.pastPurchaseSortSelected(state),
+      Selectors.pastPurchases(state)
+    );
+
+    // Since the final past purchase sort is extracted from the override request,
```

Comment:
Instead of doing this, why don't you do something like this:
```
const { sort, ...override } = overrideRequest;
```

Then you can put the sort in request and spread `override` after `request`.

Reply:
Implemented, resolving.